src/catchup/CatchupWork.cpp

```diff
@@ -88,23 +95,22 @@ CatchupWork::hasAnyLedgersToCatchupTo() const
 }
 
 void
-CatchupWork::downloadVerifyLedgerChain(CatchupRange catchupRange,
-                                       LedgerNumHashPair rangeEnd)
+CatchupWork::downloadVerifyLedgerChain(LedgerNumHashPair rangeEnd)
 {
-    auto ledgerRange = catchupRange.first;
+    // verify one extra ledger for knitting up
+    auto verifyRange = LedgerRange{mCatchupRange->mApplyLedgers.mFirst - 1,
+                                   mCatchupRange->mApplyLedgers.mLast};
     auto checkpointRange =
-        CheckpointRange{ledgerRange, mApp.getHistoryManager()};
-    CLOG(INFO, "History")
-        << "Catchup downloading ledger chain for checkpointRange ["
-        << checkpointRange.first() << ".." << checkpointRange.last() << "]";
+        CheckpointRange{verifyRange, mApp.getHistoryManager()};
+    CLOG(INFO, "History") << "Catchup downloading ledgers "
+                          << checkpointRange.toString();
     auto getLedgers = std::make_shared<BatchDownloadWork>(
         mApp, checkpointRange, HISTORY_FILE_TYPE_LEDGER, *mDownloadDir);
 
-    CLOG(INFO, "History")
-        << "Catchup verifying ledger chain for checkpointRange ["
-        << checkpointRange.first() << ".." << checkpointRange.last() << "]";
+    CLOG(INFO, "History") << "Catchup verifying ledgers "
```

Comment:
That may be harder to do that is seems. I mean, it is easy for verifying ledgers and applying transaction, but similar pattern also occurs in `downloadVerifyLedgerChain`, which uses generic `BatchDownloadWork`.

I think we can either:
- extract specific work classes for those cases - expensive, probably not worth it
- add string paramter to BasicWork class that gets logged when work first runs `onRun` (there was `onStart` at some time) - generic
- assume that works are scheduled just after they are created in CatchupWork (which I think is the case), and leave it as it is

Reply:
why not just do it in `onReset` methods of child classes (VerifyLedgerChain, etc)? It is called during transition PENDING->RUNNING, which seems like a great place to log the initial message.

Reply:
(PENDING is internal to BasicWork, but you can add a small method `bool BasicWork::started() const`)

Reply:
The problem is that we do not control all child classes at this point (as `downloadVerifyLedgerChain` uses generic `BatchDownloadWork`), so just adding logging to `onReset` won't help.

Also the documentation for `onReset` says `onReset is expected to restore work's state to its initial condition; this includes cleaning up side effects (e.g. produced files), restoring member variables etc.`. There is nothing about it being used on start of application work, so I think that either the name is wrong, documentation, or the design (I don't like the idea of work being reset at all, I would prefer the object to be recreated from scratch, but that is topic for another discussion on another time).

Reply:
I'm not sure what you mean: `BatchDownloadWork` knows what type of files it is downloading (see mFileType member variable), so its own logging should be sufficient. We shouldn't be passing log strings in the constructor - each work already has enough information on what it is doing to produce a log. 

Regarding `onReset`, there are many reasons why it is used instead of recreating objects, one being that work state is not completely wiped out during a reset (otherwise, functionality like retries would not work since it needs to remember state). This is outside of the scope of this PR, but I'm happy to discuss it elsewhere if you feel strongly about this.

Reply:
OK, you are right. I don't really care about the `BasicWork` inside `downloadVerifyLedgerChain`, as changing `BatchDownloadWork` and `VerifyLedgerChainWork` would be enough. Nice!

As for Work keeping track on how many times it was restarted - it starts looking to me as it is too much responsibility for one class (but it is so low-priority that we will probably never get back to that topic anyway).